listenbrainz/webserver/static/js/src/recent/RecentListens.tsx

```diff
@@ -0,0 +1,306 @@
+        <div className="row">
+          <div className="col-md-8">
+            {!listens.length && (
+              <div className="lead text-center">
+                <p>No listens yet</p>
+              </div>
+            )}
+            {listens.length > 0 && (
+              <>
+                <div id="listens">
+                  {listens.map((listen) => {
+                    const recordingMBID = getRecordingMBID(listen);
+                    const artistMBIDs = getArtistMBIDs(listen);
+                    const trackMBID = get(
+                      listen,
+                      "track_metadata.additional_info.track_mbid"
+                    );
+                    const releaseGroupMBID = getReleaseGroupMBID(listen);
+
+                    const isListenReviewable =
+                      Boolean(recordingMBID) ||
+                      artistMBIDs?.length ||
+                      Boolean(trackMBID) ||
+                      Boolean(releaseGroupMBID);
+                    /* eslint-disable react/jsx-no-bind */
+                    const additionalMenuItems = (
+                      <>
+                        <ListenControl
+                          title="Pin this recording"
+                          icon={faThumbtack}
+                          action={this.updateRecordingToPin.bind(this, listen)}
+                          dataToggle="modal"
+                          dataTarget="#PinRecordingModal"
+                        />
+                        {isListenReviewable && (
+                          <ListenControl
+                            title="Write a review"
+                            icon={faPencilAlt}
+                            action={this.updateRecordingToReview.bind(
+                              this,
+                              listen
+                            )}
+                            dataToggle="modal"
+                            dataTarget="#CBReviewModal"
+                          />
+                        )}
+                      </>
+                    );
+                    /* eslint-enable react/jsx-no-bind */
+                    return (
+                      <ListenCard
+                        key={`${listen.listened_at}-${listen.track_metadata?.track_name}-${listen.track_metadata?.additional_info?.recording_msid}-${listen.user_name}`}
+                        showTimestamp
+                        showUsername
+                        listen={listen}
+                        currentFeedback={this.getFeedbackForRecordingMsid(
+                          listen.track_metadata?.additional_info?.recording_msid
+                        )}
+                        updateFeedbackCallback={this.updateFeedback}
+                        newAlert={newAlert}
+                        additionalMenuItems={additionalMenuItems}
+                      />
+                    );
+                  })}
+                </div>
+                {!listens.length && (
```

Comment:
Given Line 150 in this file, this seems to be wrong. We only need this message once, no?

Reply:
Correct. And also this would never have been displayed as it was in a block displayed only if listens.length > 0